### PR TITLE
Nll loss nan

### PR DESCRIPTION
**Pull request by JackCaoG**
PT/XLA uses (`1_hot_label ` *  `logits`) to simulate the selection based on label of `nll_loss`. There are couple edge cases here we need to handle

1. inf/-inf * 0 = nan
2. nan * 0 = nan

We need to make sure that values of `logits` in non-labeled indexes would not affect the result of `nll_loss`. The solution is to replace the `NaN` in non-labeled index to 0.

We should not replace the `nan` in labeled index through.

### CODE REVIEW

**Comment by JackCaoG** [comment]
TODO: check if `BuildNllLossBackward` also needs to be updated

**Comment by JackCaoG** [comment]
I think `BuildNllLossBackward` is fine, since it does not use `reduce` on input

**Review by ailzhang — APPROVED** [review]
LGTM!
